Feature/57 remove need for recursive queries

cryptomental opened the pull request:
This PR solves #57.

I also updated copyright messages and updated Oraclize test case to test also for UpdatedLastPrice event.

pelsasser commented:
@cryptomental - this all looks great - I want to pull the changes and test it a bit locally and then will merge.  I assume this reduces deployment costs a bit and also the needed gas for our `_callback` since we are scheduling another query during the callback

pelsasser commented on contracts/oraclize/MarketContractOraclize.sol:
```diff
@@ -134,18 +111,20 @@ contract MarketContractOraclize is MarketContract, usingOraclize {
+        require(EXPIRATION - now <= 60 * 60 * 24 * 60);
+
+        bytes32 queryId = oraclize_query(
+            EXPIRATION - now,
```
I think we can use an absolute (not relative) time stamp here. 

https://docs.oraclize.it/#ethereum-quick-start-schedule-a-query-in-the-future

It sounds like using `EXPIRATION` only should work - maybe you can test and clarify?

cryptomental replied:
All right, I will test this asap.

pelsasser commented on contracts/oraclize/MarketContractOraclize.sol:
```diff
@@ -134,18 +111,20 @@ contract MarketContractOraclize is MarketContract, usingOraclize {
+        require(EXPIRATION > now);
+        // Future timestamp must be within 60 days from now.
+        // https://docs.oraclize.it/#ethereum-quick-start-schedule-a-query-in-the-future
+        require(EXPIRATION - now <= 60 * 60 * 24 * 60);
```
Can we move this to early on in the constructor perhaps?  I think that may make more sense than here.

cryptomental replied:
Will be moved to constructor.

pelsasser commented on contracts/oraclize/MarketContractOraclize.sol:
```diff
@@ -30,19 +29,12 @@ contract MarketContractOraclize is MarketContract, usingOraclize {
     string public ORACLE_DATA_SOURCE;
     string public ORACLE_QUERY;
-    uint public ORACLE_QUERY_REPEAT;
     uint constant public QUERY_CALLBACK_GAS = 180000;
```
I assume we can lower this now that the code path is much simpler.

cryptomental replied:
Yes, we can lower it, tests say ~115k, coverage network a bit more, so 130000 - 140000 should be a safe value.

pelsasser commented on contracts/oraclize/MarketContractOraclize.sol:
```diff
@@ -134,18 +111,20 @@ contract MarketContractOraclize is MarketContract, usingOraclize {
+        require(EXPIRATION > now);
+        // Future timestamp must be within 60 days from now.
+        // https://docs.oraclize.it/#ethereum-quick-start-schedule-a-query-in-the-future
+        require(EXPIRATION - now <= 60 * 60 * 24 * 60);
```
thoughts on using a constant variable for the 60 days value converted to seconds?

cryptomental replied:
Fine. I will create a variable for that.

cryptomental commented:
@pelsasser I added a commit that reworks the PR after your findings:

* use absolute timestamp to oraclize_query
* moved time checks to constructor
* introduced a constant variable SECONDS_PER_SIXTY_DAYS
* lowered Oraclize callback cost to 140000.
 
I hope this can be merged now.

cryptomental commented:
I am still experimenting a bit with oraclize callback gas.

cryptomental commented:
Ok, all checks passed, we can merge!
